client-server-server/src/main/scala/zio/raft/server/RaftServer.scala

```diff
@@ -0,0 +1,562 @@
+
+object RaftServer {
+  
+  /**
+   * Actions initiated by the server (internal or from Raft).
+   */
+  sealed trait ServerAction
+  
+  object ServerAction {
+    case class SendResponse(sessionId: SessionId, response: ClientResponse) extends ServerAction
+    case class SendServerRequest(sessionId: SessionId, request: ServerRequest) extends ServerAction
+    case class StepUp(sessions: Map[SessionId, SessionMetadata]) extends ServerAction
+    case class StepDown(leaderId: Option[MemberId]) extends ServerAction
+    case object Shutdown extends ServerAction
+  }
+  
+  /**
+   * Actions to forward to Raft state machine.
+   */
+  sealed trait RaftAction
+  
+  object RaftAction {
+    case class CreateSession(sessionId: SessionId, capabilities: Map[String, String]) extends RaftAction
+    case class ClientRequest(sessionId: SessionId, requestId: RequestId, payload: ByteVector) extends RaftAction
+    case class ServerRequestAck(sessionId: SessionId, requestId: RequestId) extends RaftAction
+    case class ExpireSession(sessionId: SessionId) extends RaftAction
+  }
+  
+  /**
+   * Transport abstraction for ZeroMQ SERVER socket.
+   */
+  trait ZmqServerTransport {
+    def sendMessage(routingId: RoutingId, message: ServerMessage): Task[Unit]
+    def incomingMessages: ZStream[Any, Throwable, IncomingMessage]
+  }
+  
+  case class IncomingMessage(
+    routingId: RoutingId,
+    message: ClientMessage
+  )
+  
+  case class SessionConnection(
+    routingId: Option[RoutingId],
+    expiresAt: Instant
+  )
+  
+  case class SessionMetadata(
+    capabilities: Map[String, String],
+    createdAt: Instant
+  )
+  
+  
+  /**
+   * Simplified RaftServer using pure functional state machine.
+   * 
+   * Similar to RaftClient, the server uses a unified stream that merges all events.
+   * Each ServerState knows how to handle events and transition to new states.
+   */
+  class RaftServer(
+    transport: ZmqServerTransport,
+    config: ServerConfig,
+    actionQueue: Queue[ServerAction],
+    raftActionsOut: Queue[RaftAction],
+    stateRef: Ref[ServerState]
+  ) {
+    
+    /**
+     * Stream of actions to forward to Raft state machine.
+     */
+    def raftActions: ZStream[Any, Nothing, RaftAction] =
+      ZStream.fromQueue(raftActionsOut)
+    
+    /**
+     * Submit a response from Raft back to a client.
+     */
+    def sendClientResponse(sessionId: SessionId, response: ClientResponse): UIO[Unit] =
+      actionQueue.offer(ServerAction.SendResponse(sessionId, response)).unit
+    
+    /**
+     * Send server-initiated request to a client.
+     */
+    def sendServerRequest(sessionId: SessionId, request: ServerRequest): UIO[Unit] =
+      actionQueue.offer(ServerAction.SendServerRequest(sessionId, request)).unit
+    
+    /**
+     * Confirm that a session has been committed by Raft.
+     * This is called directly by Raft and sends the response immediately.
+     */
+    def confirmSessionCreation(sessionId: SessionId): UIO[Unit] =
```

Comment:
what have you done? this should go through the stream... you had it correct the previous time.
Enqueue it as an action